processor/cumulativetodeltaprocessor/config.go

```diff
@@ -0,0 +1,56 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cumulativetodeltaprocessor
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+const (
+	// nameFieldName is the mapstructure field name for name field
+	nameFieldName = "name"
+)
+
+// Config defines the configuration for the processor.
+type Config struct {
+	config.ProcessorSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+
+	// Set of cumulative sum metrics to convert to delta
+	Metrics []Metric `mapstructure:"metrics"`
+}
+
+type Metric struct {
+	// Name of the metric. This is a required field.
+	Name string `mapstructure:"name"`
+
+	// List of resource attribte keys against which the metric will be matched
+	ResourceAttributeKeys []string `mapstructure:"resource_attribute_keys"`
+
+	// List of metric label keys against which the metric will be matched
+	MetricLabelKeys []string `mapstructure:"metric_label_keys"`
```

Comment:
Let's start with minimal configuration, I would suggest to only start with metric name until we have use-cases for others

Reply:
I saw the labels as being additional matchers that selected which data points to convert between cumulative and delta, but that the processor would "index" its memory of metrics data at full cardinality and pass through data with full cardinality. In other words, I was under the impression that we would compare all the resource attributes and metric attributes for the purposes of remembering previous state, but that we might use this line of configuration to select which series to convert. 

Now I agree with @bogdandrutu that we don't need this configuration much at all.

Reply:
Hi @bogdandrutu and @jmacd I got some confusions while writing the acutal logic. Can you help me to understand clearly (Sorry for being late here). 

Earlier I though, we will use a given set of resource attributes and metric attributes along with the metric name to uniquely identify a metric data point to store the previous state. If we don't take these attributes and want to match against all properties, I see two problems,

1. Matching against all properties will require more space in memory/map. We may have some high cardinality attributes and I am not sure how will that impact the processing.
1. I doubt this may not give expected output. Let's consider a scenario where we have three containers inside a pod. The metric name `container.memory.usage` is same but they have a metric level `container.name` which tells us about the metric source. We will store these data in a map as previous state. When the next payload will come, we will match against all the attributes. Now the concern is, if we get any attribute other than `container.name` which has different value in the next paylaod for the same container, then we will fail to update the previous state. How can we overcome this? I am not sure if I was able to explain it clearly. If not, I can add additional example.

Waiting for your suggestions. Thanks.